qiskit/circuit/equivalence.py

```diff
@@ -37,11 +35,53 @@ def __init__(self, *, base=None):
 
         Args:
             base (Optional[EquivalenceLibrary]):  Base equivalence library to
-                will be referenced if an entry is not found in this library.
+                be referenced if an entry is not found in this library.
         """
         self._base = base
 
-        self._map = {}
+        if base is None:
+            self._graph = rx.PyDiGraph()
+            self._key_to_node_index = dict()
+            self._num_gates_for_rule = dict()
+            self._rule_count = 0
+        else:
+            self._graph = copy.deepcopy(base._graph)
+            self._key_to_node_index = copy.deepcopy(base._key_to_node_index)
+            self._num_gates_for_rule = copy.copy(base._num_gates_for_rule)
+            self._rule_count = base._rule_count
+
+    @property
+    def graph(self) -> rx.PyDiGraph:
+        """Return graph representing the equivalence library data.
+
+        Returns:
+            PyDiGraph: A graph object with equivalence data in each node.
+        """
+        return self._graph
+
+    @property
+    def key_to_node_index(self) -> dict:
+        """Return map of Keys to node indices
```

Comment:
```suggestion
        """Return map of Keys to graph node indices
```

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/8790/commits/837b18558b075631bed822f4c3ba2cee13b98b87.